### PR TITLE
Process data

### PR DESCRIPTION
**Pull request by tarikbouari**
In this milestone, I implement the following tasks
# PR description

- Create models based on the ERD diagram
-  Insert data in the tables
- Create Users, Posts, Comments, Likes
- Create different methods
- A method that returns the 3 most recent posts for a given user.
- A method that updates the posts counter for a user.
- A method that returns the 5 most recent comments for a given post.
- A method that updates the comments counter for a post.
- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (KDlamini) [inline app/models/user.rb]: Kindly include foreign keys on your associations to referrence the author using `author_id`.

```
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :user
```
Comment (KDlamini) [inline app/models/post.rb]: Kindly use `author` for the association then reference the `User` class. 

`belongs_to :author, class_name: 'User'`

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (KDlamini) [inline app/models/like.rb]: Same here, please implement your association using this approach `belongs_to :author, class_name: 'User'`

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (KDlamini) [inline app/models/comment.rb]: And here, please implement your association using this approach `belongs_to :author, class_name: 'User'`

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  has_many :comments
+  has_many :likes
+
+  after_save :posts_count
+
```
Comment (KDlamini) [inline app/models/post.rb:7]: Well done adding the `after_save` Rails Active Callback to call the `#update_post_counter` method every time a new post has been saved to the database. 👏🏾👏🏾👏🏾

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @tarikbouari 🙋‍♀️,  

Great work so far! 💯 
You have done a great job processing data in models for the blog app. 🎊 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Correct PR title
✔️  Correct use of Gitflow 
✔️  Commit messages are meaningful
✔️  No linter errors

## Required Changes ♻️
 
Well done documenting your README 👏🏾. Please brush up on a few things to make it more professional.

- Under "Getting Started" please include all the subheadings (setup, run tests, and deployment). Do not delete any section on the README except for the FAQ section since it is optional. Kindly mention how to clone your repo under `setup`. You can leave a `none` text under `run tests` and `deployment` for now, you still need to update them in the future when more features are added.
- Since the FAQ section is optional you can delete it if you have no information to add.
- Kindly make sure that there are no broken links, please fix the link to your license.

 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @tarikbouari  👋 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omarsalem7** [comment]
Hi @tarikbouari 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**